modules/c++/re/source/PCRE_STL.cpp

```diff
@@ -0,0 +1,219 @@
+/* =========================================================================
+ * This file is part of re-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * re-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "re/PCRE.h" // this has to come before the #ifdef checks below
+
+// we use this file if we're not using actual PCRE itself
+#if defined(USE_PCRE) && defined(__CODA_CPP11)
+
+re::PCRE::PCRE(const std::string& pattern, int flags) :
+    mPattern(pattern)
+{
+    if (!mPattern.empty())
+        compile(mPattern, flags);
+}
+
+void re::PCRE::destroy()
+{
+}
+
+re::PCRE::~PCRE()
+{
+    destroy();
+}
+
+re::PCRE::PCRE(const re::PCRE& rhs)
+{
+    mPattern = rhs.mPattern;
+    compile(mPattern);
+}
+
+re::PCRE& re::PCRE::operator=(const re::PCRE& rhs)
+{
+    if (this != &rhs)
+    {
+        destroy();
+
+        mPattern = rhs.mPattern;
+
+        compile(mPattern);
+    }
+
+    return *this;
+}
+
+re::PCRE& re::PCRE::compile(const std::string& pattern,
+                            int flags)
+{
+    mPattern = (flags==PCRE_DOTALL) ? replace_dot(pattern) : pattern;
+    try 
+    {
+        mRegex = std::regex(mPattern);
+    }
+    catch (const std::regex_error& e)
+    {
+        throw PCREException(Ctxt(FmtX("PCRE std::regex constructor error: %s",
```

Comment:
A lot of our older code uses `FmtX()` but actually try to avoid it since internally it's using a fixed-size char buffer which is just begging to overflow.  So just change to something like...

`throw PCREException(Ctxt("PCRE std::regex constructor error: " + std::string(e.what())))`